cmd/dreamboat/main.go

```diff
@@ -519,34 +518,30 @@ func ComputeDomain(domainType types.DomainType, forkVersionHex string, genesisVa
 	return types.ComputeDomain(domainType, forkVersion, genesisValidatorsRoot), nil
 }
 
-func initStreamer(ctx context.Context, cfg *config.DistributedConfig, redisClient *redis.Client, l log.Logger, m *metrics.Metrics, st stream.State) (relay.Streamer, error) {
-	timeStreamStart := time.Now()
-
-	pubsub := &redisStream.Pubsub{Redis: redisClient, Logger: l}
-
-	id := cfg.InstanceID
-	if id == "" {
-		id = uuid.NewString()
+func newStreamClient(cfg *config.DistributedConfig, redisClient *redis.Client, l log.Logger, m *metrics.Metrics, st stream.State) *stream.Client {
+	bids := &redis_stream.Topic{
+		Redis:     redisClient,
+		Logger:    l.WithField("topic", stream.BidTopic),
+		Name:      path.Join(cfg.Redis.Topic, stream.BidTopic),
+		LocalNode: cfg.LocalNode(),
 	}
 
-	streamConfig := stream.StreamConfig{
-		Logger:          l,
-		ID:              id,
-		PubsubTopic:     cfg.Redis.Topic,
-		StreamQueueSize: cfg.StreamQueueSize,
+	cache := &redis_stream.Topic{
+		Redis:     redisClient,
+		Logger:    l.WithField("topic", stream.CacheTopic),
+		Name:      path.Join(cfg.Redis.Topic, stream.CacheTopic),
+		LocalNode: cfg.LocalNode(),
 	}
 
-	redisStreamer := stream.NewClient(pubsub, st, streamConfig)
-	redisStreamer.AttachMetrics(m)
-
-	if err := redisStreamer.RunSubscriberParallel(ctx, uint(cfg.WorkerNumber)); err != nil {
-		return nil, fmt.Errorf("fail to start stream subscriber: %w", err)
+	return &stream.Client{
```

Comment:
There is a chance you may like to use the NewClient function from below

Reply:
🤦‍♂️